node/dataSources/oms.ts

```diff
@@ -0,0 +1,29 @@
+import { RequestOptions, RESTDataSource } from 'apollo-datasource-rest'
+import * as cookies from 'cookie'
+import { forEachObjIndexed } from 'ramda'
+
+
+export class OMSDataSource extends RESTDataSource<Context> {
+  public lastUserOrder = () => this.get(`/user/orders/last`)
+
+  get baseURL() {
+    const { vtex: { account } } = this.context
+
+    return `http://${account}.vtexcommercestable.com.br/api/oms`
+  }
+
+  protected willSendRequest(request: RequestOptions) {
+    const { vtex: { authToken, account }, headers:{cookie} } = this.context
+
+    const parsedCookie = cookies.parse(cookie)
```

Comment:
I'm don't think that this is the cookie that you should send. The logged in cookie is the `VtexIdClientAutCookie__{account}`. Also dunno if you should care also for the case when the user is impersonated by a call center operator. In any case, the backend should take care of this validations for you and you could just pass the whole cookie from the request to your own request.

Reply:
fixed